src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,111 @@
+     * contains a date in a recognisable format. If it does, the information is also checked if it contains a time
+     * in a recognisable, 24-hour format. It then stores those accordingly. The rest of the string is stored as is.
+     *
+     * @param name The name of the new Deadline task.
+     * @param dueStr The information on when the Deadline task is due.
+     */
+    public Deadline(String name, String dueStr) {
+        super(name);
+        this.dueStr = dueStr;
+
+        // Handle date and time representation, if any
+        String[] dateTimeArr = dueStr.split(" ");
+        try {
+            due = LocalDate.parse(dateTimeArr[0]);
+        }
+        catch (DateTimeParseException e) {
+            hasDate = false;
+            hasTime = false;
+        }
+        if (hasDate) {
+            if (dateTimeArr.length >= 2) {
+                int time = -1;
+                try {
+                    time = Integer.parseInt(dateTimeArr[1]);
+                }
+                catch (NumberFormatException e) {
+                    hasTime = false;
+                }
+                if (hasTime && 0 <= time && time <= 2359) {
+                    timeStr = dateTimeArr[1];
+                } else {
+                    hasTime = false;
+                }
+            } else {
+                hasTime = false;
+            }
+        }
+
+        // Handle representation of information based on whether there is a recognisable date and time
+        int startIndex = hasTime ? 2 : hasDate ? 1 : 0;
+        this.dueStr = "";
+        for (int i = startIndex; i < dateTimeArr.length; i++) {
+            if (i != 0) this.dueStr += " ";
+            this.dueStr += dateTimeArr[i];
+        }
+        if (hasTime) {
+            int hours = Integer.parseInt(timeStr.substring(0, 2));
+            int mins = Integer.parseInt(timeStr.substring(2));
+            String mornAftStr = "";
+            if (hours > 12) {
+                hours -= 12;
+                mornAftStr = "pm";
+            } else {
+                if (hours == 0) {
+                    hours = 12;
+                }
+                mornAftStr = "am";
+            }
+            String minPrependStr = "";
+            if (mins < 10) {
+                minPrependStr = "0";
+            }
+            timeStr = hours + ":" + minPrependStr + mins + mornAftStr;
+            this.dueStr = " " + timeStr + this.dueStr;
+        }
+        if (hasDate) {
+            this.dueStr = due.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + this.dueStr;
+        }
+    }
+
+    /**
+     * Converts the Deadline task to the format used for saving in the storage file on the user's computer.
+     *
+     * @return The save format of the Deadline task.
+     */
+    public String convertToSaveFormat() {
+        return "D|" + super.convertToSaveFormat() + "|" + dueStr;
+    }
+
+    /**
+     * {@inheritDoc}
```

Comment:
Is ```@inheritDoc``` is part of the coding standard?

Reply:
Yup wasn't really sure about this but my TA mentioned that there was a way to inherit the JavaDoc comments from superclass methods, so I googled and found this!